src/AnalysisPrograms/Recognizers/PteropusSpecies.cs

```diff
@@ -0,0 +1,450 @@
+            }
+
+            //Set following true if you want special debug spectrogram, i.e. with special plots
+            //In addition, standard spectrograms are produced when you set true in the config file, Towsey.PteropusSpecies.yml.
+            if (false)
+            {
+                //var image = sonogram.GetImageFullyAnnotated("Test");
+                string speciesName = genericConfig[AnalysisKeys.SpeciesName] ?? "Pteropus species";
+                var image = SpectrogramTools.GetSonogramPlusCharts(results1.Sonogram, results1.Events, results1.Plots, null);
+                var opPath = outputDirectory.Combine(FilenameHelpers.AnalysisResultName(Path.GetFileNameWithoutExtension(audioRecording.BaseName), speciesName, "png", "DebugSpectrogram"));
+                string imageFilename = audioRecording.BaseName + ".profile.png";
+                image.Save(Path.Combine(outputDirectory.FullName, imageFilename));
+            }
+
+            return results1;
+        }
+
+        /// <summary>
+        /// THis method does the work.
+        /// </summary>
+        /// <param name="audioRecording">the recording.</param>
+        /// <param name="configuration">the config file.</param>
+        /// <param name="outputDirectory">where results are to be put.</param>
+        /// <param name="segmentStartOffset">where one segment is located in the total recording.</param>
+        /// <returns>a list of events.</returns>
+        internal static RecognizerResults TerritorialCall(AudioRecording audioRecording, Config configuration, Config profile, DirectoryInfo outputDirectory, TimeSpan segmentStartOffset)
+        {
+            // get the common properties
+            string speciesName = configuration[AnalysisKeys.SpeciesName] ?? "Pteropus species";
+            string abbreviatedSpeciesName = configuration[AnalysisKeys.AbbreviatedSpeciesName] ?? "Pteropus";
+
+            // The following parameters worked well on a ten minute recording containing 14-16 calls.
+            // Note: if you lower the dB threshold, you need to increase maxDurationSeconds
+            int minHz = profile.GetIntOrNull(AnalysisKeys.MinHz) ?? 800;
+            int maxHz = profile.GetIntOrNull(AnalysisKeys.MaxHz) ?? 8000;
+            double minDurationSeconds = profile.GetDoubleOrNull(AnalysisKeys.MinDuration) ?? 0.15;
+            double maxDurationSeconds = profile.GetDoubleOrNull(AnalysisKeys.MaxDuration) ?? 0.5;
+            double decibelThreshold = profile.GetDoubleOrNull(AnalysisKeys.DecibelThreshold) ?? 9.0;
+
+            var minTimeSpan = TimeSpan.FromSeconds(minDurationSeconds);
+            var maxTimeSpan = TimeSpan.FromSeconds(maxDurationSeconds);
+
+            //######################
+            //2.Convert each segment to a spectrogram. Don't use samples in this recogniser.
+            //var samples = audioRecording.WavReader.Samples;
+
+            // make a spectrogram
+            var sonoConfig = new SonogramConfig
```

Comment:
the spectrogram creation looks identical to the one Wingbeats method. If it is identical, factor it out so we don't repeat the calculation twice